lamindb_setup/core/upath.py

```diff
@@ -548,3 +548,15 @@ def get_stat_dir_gs(path: UPath) -> Tuple[int, str, str, int]:
     n_objects = len(md5s)
     hash, hash_type = hash_md5s_from_dir(md5s)
     return sum(sizes), hash, hash_type, n_objects
+
+
+def check_s3_storage_location_empty(path: UPath) -> bool:
+    objects = path.fs.find(path.as_posix(), detail=True)
+    n_objects = len(objects.values())
+    if n_objects == 0:
+        return True
+    else:
+        raise ValueError(
```

Comment:
The way this function is called and typed should rather make it return False here than raising an error.

Reply:
Since this is supposed to be a gating function, I'll keep it as is (raising error if storage is not empty) and return None if the opposite holds. Will type it accordingly.